pkg/models/monitoring/monitoring.go

```diff
@@ -38,14 +40,22 @@ func NewMonitoringOperator(client monitoring.Interface) MonitoringOperator {
 	return &monitoringOperator{client}
 }
 
-// TODO(huanggze): reserve for custom monitoring
-func (mo monitoringOperator) GetMetrics(stmts []string, time time.Time) Metrics {
-	panic("implement me")
+func (mo monitoringOperator) GetMetric(expr, namespace string, time time.Time) (monitoring.Metric, error) {
+	// Enforce namespace matcher
+	expr, err := expressions.ReplaceNamespaceFns["prometheus"](expr, namespace)
+	if err != nil {
+		return monitoring.Metric{}, err
+	}
+	return mo.c.GetMetric(expr, time), nil
 }
 
-// TODO(huanggze): reserve for custom monitoring
-func (mo monitoringOperator) GetMetricsOverTime(stmts []string, start, end time.Time, step time.Duration) Metrics {
-	panic("implement me")
+func (mo monitoringOperator) GetMetricOverTime(expr, namespace string, start, end time.Time, step time.Duration) (monitoring.Metric, error) {
+	// Enforce namespace matcher
+	expr, err := expressions.ReplaceNamespaceFns["prometheus"](expr, namespace)
```

Comment:
Same here, it's better to check if the value of key "prometheus" exist first, and then call the label replace function?